src/goal.rs

```diff
@@ -97,11 +102,10 @@ impl Goal {
             //assign slots that are within the specified after_time and before_time
             let slots = slot_generator(self.after_time, self.before_time, &time_period);
             //calculate flexibility
-            let mut hours_available = 0;
+            let mut flexibility = 0;
             for slot in &slots {
-                hours_available += slot.num_hours();
+                flexibility += slot.num_hours() - self.duration + 1;
```

Comment:
```suggestion
                flexibility += std::cmp::max(0, slot.num_hours() - self.duration + 1);
```

Reply:
The `slot_generator` only assigns valid slots for the task so I don't think `slot.num_hours() - self.duration + 1` will be less than 0. If it doesn't wouldn't we rather that the code to panics?

Reply:
You're absolutely right!